lib/backend/test/suite.go

```diff
@@ -619,6 +619,17 @@ func testEvents(t *testing.T, newBackend Constructor) {
 	_, err = uut.Get(ctx, item.Key)
 	require.Error(t, err)
 
+	// A backend may implement the ttldeleter interface
+	// to do synchronous TTL deletion during testing.
+	type ttldeleter interface {
+		TTLDelete(context.Context) error
+	}
+
+	if ttl, ok := uut.(ttldeleter); ok {
+		err := ttl.TTLDelete(ctx)
+		require.NoError(t, err)
+	}
```

Comment:
With the crdb ttl cron job, the fastest it will run is every minute. Even doing some hacky internal stuff like `UPDATE system.scheduled_jobs SET next_run = now() WHERE schedule_name like 'row-level-ttl-%';`. It seems like internally crdb only checks for jobs to run every minute. So although the job could get scheduled any time in the next 0-60 seconds the worst case is still waiting the full 60 seconds.

Reply:
We could but why build it if its already supported by the database? I am pretty sure we use the database ttl deletion for other backends that support it.

Does the dynamodb backend not hit this same issue?

Reply:
The dynamodb backend deletes expired items on get (and tests deliberately trigger that to pass), which is a seriously weird behavior that we should absolutely not take as an example of what to do.

Reply:
I'm just not sure passing this one test case justifies adding application side ttl deletion.

Some other options are skipping this specific scenario for or setting a longer timeout on this test for crdb.

Reply:
It would be nice to have the option to manually run the test suite with the native TTL deletion set as fast as possible and the timeouts set very very long.

Reply:
Checkout [87402e9](https://github.com/gravitational/teleport/pull/36362/commits/87402e99a24e56d478a58be3d092ccc0694573f2). I just added an env var to set the timeout.